### PR TITLE
Api version kwargs

### PR DESCRIPTION
**Pull request by jorgii**
Add user selected version (v3 by default) in Client via kwargs dict

### CODE REVIEW

File: test.py [hunk 1]
```diff
@@ -2,43 +2,53 @@
...
-        result = mj.contact.get(filter={'id': 5771382}).json()
+        result = self.client.contact.get(filter={'id': 5771382}).json()
         self.failUnless('Count' in result)
 
     def test_post_with_no_param(self):
-        result = mj.sender.create(data={}).json()
+        result = self.client.sender.create(data={}).json()
         self.failUnless('StatusCode' in result and result['StatusCode'] is not 400)
 
+    def test_clinet_custom_version(self):
```
Comment (arnaudbreton) [inline test.py]: @jorgii typo, `clinet` => `client`

File: mailjet_rest/client.py [hunk 2]
```diff
@@ -9,12 +9,19 @@
 
 requests.packages.urllib3.disable_warnings()
 
+
 class Config(object):
-    API_URL = 'https://api.mailjet.com/v3/'
-    API_DOC = 'http://dev.mailjet.com/email-api/v3/'
+    API_URL = 'https://api.mailjet.com/'
+    API_DOC = 'http://dev.mailjet.com/email-api/'
+    version = 'v3/'  # urljoin requires forward slash
```
Comment (arnaudbreton) [inline mailjet_rest/client.py]: @jorgii as per our discussions, the trailing `/` shouldn't be part of the version but added when building the URL

File: mailjet_rest/client.py [hunk 3]
```diff
@@ -9,12 +9,19 @@
 
 requests.packages.urllib3.disable_warnings()
 
+
 class Config(object):
-    API_URL = 'https://api.mailjet.com/v3/'
-    API_DOC = 'http://dev.mailjet.com/email-api/v3/'
+    API_URL = 'https://api.mailjet.com/'
+    API_DOC = 'http://dev.mailjet.com/email-api/'
```
Comment (arnaudbreton) [inline mailjet_rest/client.py]: API Reference URL to be accurate. Docs is made of Guides+Reference.

**Review by arnaudbreton — CHANGES_REQUESTED** [review]

File: test.py (hunk 1 above)
Comment (jorgii) [inline test.py]: Fixed in 051f8cb

File: mailjet_rest/client.py (hunk 3 above)
Comment (jorgii) [inline mailjet_rest/client.py]: Changed in 8132e10

File: mailjet_rest/client.py (hunk 2 above)
Comment (jorgii) [inline mailjet_rest/client.py]: Done in 1c74433. Also README.md updated accordingly at 374f126

File: README.md [hunk 4]
```diff
@@ -36,10 +36,12 @@ import os
 API_KEY = os.environ['MJ_APIKEY_PUBLIC']
 API_SECRET = os.environ['MJ_APIKEY_PRIVATE']
 
-mailjet = Client(auth=(API_KEY, API_SECRET))
+mailjet = Client(auth=(API_KEY, API_SECRET), version='v3/')
```
Comment (arnaudbreton) [inline README.md]: `version` should now be `v3`?

File: mailjet_rest/client.py [hunk 5]
```diff
@@ -9,12 +9,21 @@
...
+    version = 'v3'
+
+    def __init__(self, version=None):
+        if version is not None:
+            self.version = version
 
     def __getitem__(self, key):
         url = self.API_URL[0:]
+        # Append version to URL.
+        # Forwart slash is ignored if present in self.version.
```
Comment (arnaudbreton) [inline mailjet_rest/client.py]: Typo: `forwart` => `forward`

File: test.py [hunk 6]
```diff
@@ -2,43 +2,57 @@
...
         self.failUnless('StatusCode' in result and result['StatusCode'] is not 400)
 
+    def test_client_custom_version(self):
+        self.client = Client(
+            auth=self.auth,
+            version='v3.1'
+        )
+        self.assertEqual(self.client.config.version, 'v3.1')
+        self.assertEqual(
+            self.client.config['contact'][0],
```
Comment (arnaudbreton) [inline test.py]: Not the best example, knowing `contact` isn't a `v3.1` resource yet.

**Review by arnaudbreton — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (jorgii) [inline README.md]: Fixed in 3dfe052

File: mailjet_rest/client.py (hunk 5 above)
Comment (jorgii) [inline mailjet_rest/client.py]: fixed in edaa16b

File: test.py (hunk 6 above)
Comment (jorgii) [inline test.py]: Added better example in 3f28c50

**Pull request merged**